server/channels/app/report.go

```diff
@@ -4,11 +4,136 @@
 package app
 
 import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
 	"net/http"
 
 	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/i18n"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+	"github.com/mattermost/mattermost/server/v8/platform/shared/filestore"
 )
 
+func (a *App) SaveReportChunk(format string, prefix string, count int, reportData []model.ReportableObject) *model.AppError {
+	switch format {
+	case "csv":
+		return a.saveCSVChunk(prefix, count, reportData)
+	}
+	return model.NewAppError("SaveReportChunk", "app.save_report_chunk.unsupported_format", nil, "unsupported report format", http.StatusBadRequest)
+}
+
+func (a *App) saveCSVChunk(prefix string, count int, reportData []model.ReportableObject) *model.AppError {
+	var buf bytes.Buffer
+	w := csv.NewWriter(&buf)
+
+	for _, report := range reportData {
+		err := w.Write(report.ToReport())
+		if err != nil {
+			return model.NewAppError("saveCSVChunk", "app.save_csv_chunk.write_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+
+	w.Flush()
```

Comment:
As per [documentation](https://pkg.go.dev/encoding/csv#Writer.Flush), it looks like might need to check for an error from `w.Flush()`:

> To check if an error occurred during the Flush, call Error.